### PR TITLE
Doc Blocks: Add support for `of` prop to `Primary` block

### PR DESCRIPTION
**Pull request by Wilson2k**
Partially implements #22490 

## What I did

- Add support for `of` prop via useOf with "meta" as only valid type
- Added stories
- Updated documentation, using #22552 as an example

## How to test

1. Run `yarn storybook:blocks`
2. Open Storybook in your browser
3. Access Primary stories

## Checklist

- [x ] Make sure your changes are tested (stories and/or unit, integration, or end-to-end tests)
- [x ] Make sure to add/update documentation regarding your changes
- [x ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "build", "documentation", "maintenance", "dependencies", "other"]`

### CODE REVIEW

**Comment by Wilson2k** [comment]
Hi everyone, this is my first contribution to the codebase. Please let me know if I can improve anything, thanks!

File: code/ui/blocks/src/blocks/Primary.stories.tsx
Comment (kylegach) [inline code/ui/blocks/src/blocks/Primary.stories.tsx:6]: Curious why you changed from the `satisfies` operator to direct assignment of the type?

File: docs/api/doc-block-primary.md
Comment (kylegach) [inline docs/api/doc-block-primary.md:36]: Can we state something similar to [Meta's `of` prop description](https://storybook.js.org/docs/react/api/doc-block-meta#of)?

```suggestion
Specifies which CSF file is used to find the first story, which is then rendered by this block. Pass the full set of exports from the CSF file (not the default export!).
```

File: code/ui/blocks/src/blocks/Primary.stories.tsx
Comment (Wilson2k) [reply code/ui/blocks/src/blocks/Primary.stories.tsx:6]: Hi, thanks for reviewing my code! I changed it based off other PRs and off of Description.stories.tsx. But this is a good question, and after some research, I'll change it back to satisfies since it's just better for validation and type safety.

File: docs/api/doc-block-primary.md
Comment (Wilson2k) [reply docs/api/doc-block-primary.md:36]: Yeah sure thing, I'll push a commit with the other fix today.

**Review by jonniebigodes — APPROVED** [review]
@Wilson2k, thanks for addressing the feedback so promptly. Appreciate it 🙏 ! Documentation-wise, it's all good on my end. I'll defer to the other maintainers to see if they have anything else to say about this great contribution.


cc @JReinhold 
Hope you have a great day.

Stay safe

**Comment by yannbf** [comment]
Hey @JReinhold could you please take a look?

File: code/ui/blocks/src/blocks/Primary.tsx
Comment (JReinhold) [inline code/ui/blocks/src/blocks/Primary.tsx:52]: The second argument to `useOf` is an array of valid types, here it would make sense to specify `'meta'` as the only valid type.

```suggestion
    const resolvedOf = useOf(of || 'meta', ['meta']);
```

File: MIGRATION.md
Comment (JReinhold) [inline MIGRATION.md:1510]: These migration guides are between specific versions, adding this note here communicates that this was introduced in 7.0, while it's probably being introduced in 7.6.

I suggest you create a new top section below the Table Of Content called "From version 7.5.0 to 7.6.0", that includes this content. Given that this will move it "out" of the doc block section, you probably need to add a bit more context and link to this doc block section.

**Review by JReinhold — CHANGES_REQUESTED** [review]
Thanks for the PR and the patience, generally looks good! A few minor changes needed, but then it's ready!

File: code/ui/blocks/src/blocks/Primary.tsx
Comment (Wilson2k) [reply code/ui/blocks/src/blocks/Primary.tsx:52]: Ok thanks for your review, good catch! I'll push a commit asap.

File: MIGRATION.md
Comment (Wilson2k) [reply MIGRATION.md:1510]: Ok sounds good, I'll push a commit along with the other one updating the docs.

File: code/ui/blocks/src/blocks/Primary.tsx
Comment (JReinhold) [inline code/ui/blocks/src/blocks/Primary.tsx:53]: I'm sorry for not realising this sooner, I just thought of it now. I think now that we supply `'meta'` as the only valid type, we don't need the "complex" error handling in `getPrimaryFromResolvedOf`, as that is already done within `useOf` based on the second argument - it should even be typed correctly.

So maybe you could try this and remove `getPrimaryFromResolvedOf` entirely?

```suggestion
    const resolvedOf = useOf(of || 'meta', ['meta']);
    story = resolvedOf.csfFile.stories[0] || null;
```

Comment (Wilson2k) [reply code/ui/blocks/src/blocks/Primary.tsx:53]: Hey, sorry for the late reply, I just tested it and it seems to work fine. Good catch, I'll push a commit with the changes.

**Review by JReinhold — APPROVED** [review]

**Pull request merged**
